taui/src/components/form.js

```diff
@@ -158,13 +175,13 @@ class Form extends React.PureComponent<Props> {
             className='map-sidebar__select'
             clearable={false}
             filterOptions={destinationFilterOptions}
-            options={locations}
+            options={locationsWithLabels}
             optionHeight={SELECT_OPTION_HEIGHT}
             onChange={this.selectDestination}
             placeholder={t('Geocoding.StartPlaceholder')}
             style={SELECT_STYLE}
             wrapperStyle={SELECT_WRAPPER_STYLE}
-            value={destination}
+            value={{...destination, label: t(destination.value) + ': ' + destination.label}}
```

Comment:
This is a good opportunity for cleanup of the existing code -- the [`react-select` docs](https://github.com/JedWatson/react-select/tree/v1.x#usage) indicate that `value` should be "either a string or number" and should get passed to the `Select` element whenever the selected option changes, so I think it should be safe to use just `destination.value` here:
```suggestion
            value={destination.value}
```